### PR TITLE
Gilded rose PR 

### PR DESCRIPTION
**Pull request by ClaraBollenGandolfo**
# READ ME BEFORE SUBMITTING A PR

Please do not submit a PR with your solution to the Gilded Rose Kata. This repo is intended to be used as a starting point for the kata.

- [ ] I acknowledge that this PR is not a solution to the Gilded Rose Kata, but an improvement to the template.
- [ ] I acknowledge that I have read [CONTRIBUTING.md](https://github.com/emilybache/GildedRose-Refactoring-Kata/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: Kotlin/local.properties
Comment (AmeliaIsCoding) [inline Kotlin/local.properties:1]: Take a look at the contents of this file - we don't want to check it into git. 
see if you can find some documentation on .gitignore to help you figure out how to make sure this file doesn't get committed!

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 1]
```diff
@@ -2,57 +2,69 @@ package com.gildedrose
...
-                    if (items[i].name != "Backstage passes to a TAFKAL80ETC concert") {
-                        if (items[i].quality > 0) {
-                            if (items[i].name != "Sulfuras, Hand of Ragnaros") {
-                                items[i].quality = items[i].quality - 1
-                            }
-                        }
-                    } else {
-                        items[i].quality = items[i].quality - items[i].quality
-                    }
+            if (items[i].name == "+5 Dexterity Vest" || items[i].name == "Elixir og the Mongoose") {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: I think the intention here is _any_ other items should follow these rules, not just the vest / elixer.

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 2]
```diff
@@ -2,57 +2,69 @@ package com.gildedrose
...
+
+                if (items[i].sellIn < 0) {
+                    items[i].quality -= 2
+                } else {
+                    items[i].quality -= 1
+                }
+            }
+
+            if (items[i].name == "Conjured Mana Cake") {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: the intention here is if an item name starts with "Conjured", it should follow these rules -> not just mana cake!

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 3]
```diff
@@ -2,57 +2,69 @@ package com.gildedrose
...
+                    items[i].quality -= 4
                 } else {
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1
-                    }
+                    items[i].quality -= 2
                 }
             }
+
+            if (items[i].name == "Elixir of the Mongoose") {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: it looks like this block is effectively a repeat from one above

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 4]
```diff
@@ -2,57 +2,69 @@ package com.gildedrose
...
 
+    //
     fun updateQuality() {
         for (i in items.indices) {
-            if (items[i].name != "Aged Brie" && items[i].name != "Backstage passes to a TAFKAL80ETC concert") {
-                if (items[i].quality > 0) {
-                    if (items[i].name != "Sulfuras, Hand of Ragnaros") {
-                        items[i].quality = items[i].quality - 1
-                    }
+            if (items[i].name != "Sulfuras, Hand of Ragnaros") {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: you could use a `when` statement instead of `if` and `if else`.
please see https://kotlinlang.org/docs/control-flow.html#when-expressions-and-statements

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 5]
```diff
@@ -2,57 +2,69 @@ package com.gildedrose
 
 class GildedRose(val items: List<Item>) {
 
+    //
     fun updateQuality() {
         for (i in items.indices) {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: you could use a `forEach` loop here:
```
items.forEach { 
   ///code here
 }
```

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 6]
```diff
@@ -6,14 +6,161 @@ import org.junit.jupiter.api.Test
 internal class GildedRoseTest {
 
     @Test
-    fun foo() {
-        val items = listOf(Item("foo", 0, 0))
+    fun dexterityVestNormalBehaviour() {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: test names:

- you can write test names as strings by using backticks, eg:
```
@Test
fun `test name here`() {
//test code here
}
```
- from a test name, the expected behaviour should be evident, so don't use vagueries such as 'normal behaviour' - another developer doesn't know what the normal behaviour should be!

- test names should be easy to say aloud / use normal english

for example, this test would be better named "Dextery vest quality decreases by one before expiry"

have a go at renaming all the tests :)

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 7]
```diff
@@ -6,14 +6,161 @@ import org.junit.jupiter.api.Test
...
+        val items = listOf(Item("+5 Dexterity Vest", 10, 20))
         val app = GildedRose(items)
         app.updateQuality()
-        assertEquals("fixme", app.items[0].name)
+        assertEquals("+5 Dexterity Vest",  app.items[0].name)
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: it's better practice to split up the assertions into different tests. so you'd have 3 tests here rather than one - each test is testing a different thing.

if you feel that you don't need name / sell in tests for all the cases, then you can get rid of those tests! make sense?

File: Kotlin/gradle/wrapper/gradle-wrapper.properties
Comment (AmeliaIsCoding) [inline Kotlin/gradle/wrapper/gradle-wrapper.properties:1]: Extension if you want to: 
The rules for the different cases are currently embedded into the `updateQuality` function. This isn't ideal as they don't really belong there - each code entity should have one responsibility. So right now `updateQuality` has two responsibilities - storing the rules for each case, and also performing the quality / sellIn updates. 
See single responsibility principle. https://stackify.com/solid-design-principles/

You could make classes to store the separate rules for each case. 

Hint (just one possible approach):

```
abstract class UpdatableItem(val item: Item) {
    abstract fun update()

    fun decreaseSellIn() {
        item.sellIn -= 1
    }

    fun increaseQuality(amount: Int = 1) {
        item.quality = minOf(50, item.quality + amount)
    }

    fun decreaseQuality(amount: Int = 1) {
        item.quality = maxOf(0, item.quality - amount)
    }
}

class NormalItem(item: Item) : UpdatableItem(item) {
    override fun update() {
        decreaseSellIn()
        decreaseQuality()
        if (item.sellIn < 0) decreaseQuality()
    }
}
```

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 8]
```diff
@@ -6,14 +6,161 @@ import org.junit.jupiter.api.Test
...
+        val items = listOf(Item("Backstage passes to a TAFKAL80ETC concert", 5, 30))
+        val app = GildedRose(items)
+        app.updateQuality()
+        assertEquals("Backstage passes to a TAFKAL80ETC concert",  app.items[0].name)
+        assertEquals(4, app.items[0].sellIn)
+        assertEquals(33, app.items[0].quality)
+    }
+
+    @Test
+    fun  backstagePassesAt0Behaviour() {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: what about after the sell in date? should we test this case too?

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 9]
```diff
@@ -6,14 +6,161 @@ import org.junit.jupiter.api.Test
...
+        val items = listOf(Item("+5 Dexterity Vest", 0, 20))
+        val app = GildedRose(items)
+        app.updateQuality()
+        assertEquals("+5 Dexterity Vest",  app.items[0].name)
+        assertEquals(-1, app.items[0].sellIn)
+        assertEquals(18, app.items[0].quality)
+    }
+
+    @Test
+    fun agedBrieNormalBehaviour() {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: what about after the expiry date? should we test this behaviour too?

**Review by AmeliaIsCoding — COMMENTED** [review]
Well done! You've got a good grasp of how testing works. There's a few improvements to make to the tests - there are some cases that you haven't covered, and there's some naming improvements you could make. In general, try and write a test to cover each aspect of the specification, plus any further edge cases you can think of.

Your refactoring looks good, but there's some code that could be tidied up in terms of repetition. There's also some Kotlin syntax that you can use that will make your life a bit easier - see the comments! :) 
As an extension, you could try and implement a class structure to hold the rules for the different classes, this would be the cleanest solution - see comments :)

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 2 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: sorry misunderstood this!

File: Kotlin/gradle/wrapper/gradle-wrapper.properties
Comment (ClaraBollenGandolfo) [reply Kotlin/gradle/wrapper/gradle-wrapper.properties:1]: I rewrote the code with this, I think it looks much better : )

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt (hunk 8 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: added

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt (hunk 9 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: added

File: Kotlin/local.properties
Comment (ClaraBollenGandolfo) [reply Kotlin/local.properties:1]: sorry, I missed this when resubmitting the PR, I've put this in place now

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 1 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: I did leave put these in as a 'depreciating good' category that can be added to but if they should be the default type I'm just realising I should check that if item.name is not one of the 'special' ones, then call decreaseQuality()

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt (hunk 6 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: done

File: Kotlin/local.properties
Comment (AmeliaIsCoding) [reply Kotlin/local.properties:1]: I'm not sure if you pushed this change as this file is still committed & I can't see any changes in the .gitignore file 👀

Comment (AmeliaIsCoding) [reply Kotlin/local.properties:1]: let me know if you want some help!

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 10]
```diff
@@ -1,58 +1,49 @@
...
+    fun decreaseQuality(amount: Int ) {
+        item.quality = maxOf(0, item.quality - amount)
+    }
+
+    fun maintainQuality() {
+        item.quality = item.quality
+        item.sellIn = item.sellIn
     }
 
+    fun QualityToZero() {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: how about `setQualityToZero` so that we can use a verb at the start?

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 11]
```diff
@@ -1,58 +1,49 @@
...
+        item.quality = 0
+    }
 }
 
+class NormalItem(item: Item) : UpdatableItem(item) {
+    override fun update() {
+        val staticGoods = arrayOf("Sulfuras, Hand of Ragnaros")
+        if (staticGoods.contains(item.name)) maintainQuality()
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: I'm not sure this is a useful function to have, why not just do nothing here?

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt [hunk 12]
```diff
@@ -13,11 +13,13 @@ fun main(args: Array<String>) {
...
             Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
-            // this conjured item does not work properly yet
             Item("Conjured Mana Cake", 3, 6)
     )
 
-    val app = GildedRose(items)
+    for (item in items) {
+
+
+    val app = NormalItem(item)
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt]: indentation seems a little wrong here btw!

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt [hunk 13]
```diff
@@ -13,11 +13,13 @@ fun main(args: Array<String>) {
             Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
             Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
             Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
-            // this conjured item does not work properly yet
             Item("Conjured Mana Cake", 3, 6)
     )
 
-    val app = GildedRose(items)
+    for (item in items) {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt:19]: in my experience `items.forEach` is preferred

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt (hunk 12 above)
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt]: what does `app` mean here? can you think of a more descriptive name?

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt [hunk 14]
```diff
@@ -13,11 +13,13 @@ fun main(args: Array<String>) {
...
+
+
+    val app = NormalItem(item)
 
     var days = 2
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt]: how does this variable get used please?

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt [hunk 15]
```diff
@@ -31,6 +33,7 @@ fun main(args: Array<String>) {
             println(item)
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt]: is this just here for debugging purposes? not super important, but for the future best to avoid committing code that isn't intended to be permanent

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 16]
```diff
@@ -6,14 +6,178 @@ import org.junit.jupiter.api.Test
...
-    fun foo() {
-        val items = listOf(Item("foo", 0, 0))
-        val app = GildedRose(items)
-        app.updateQuality()
-        assertEquals("fixme", app.items[0].name)
+    fun `Dexterity vest quality decreases by one before expiry`() {
+        val item = Item("+5 Dexterity Vest", 10, 20)
+        val app = NormalItem(item)
+        app.update()
+        assertEquals("+5 Dexterity Vest", app.item.name)
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt:13]: is best to have these assertions split into separate tests. If you then think you don't need that many tests for the name and the sell in, you could remove some :)

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 17]
```diff
@@ -6,14 +6,178 @@ import org.junit.jupiter.api.Test
...
+        val item = (Item("Aged Brie", 0, 0))
+        val app = NormalItem(item)
+        app.update()
+        assertEquals("Aged Brie", app.item.name)
+        assertEquals(-1, app.item.sellIn)
+        assertEquals(1, app.item.quality)
+    }
+
+    @Test
+    fun `Elixir of the mongoose decreases in quality by one before expiry`() {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt:61]: is it important here that it's elixir of the mongoose? I think a better test name would be

`Regular item decreases in quality by one before expiry`

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 18]
```diff
@@ -1,58 +1,49 @@
...
+        item.quality = item.quality
+        item.sellIn = item.sellIn
     }
 
+    fun QualityToZero() {
+        item.quality = 0
+    }
 }
 
+class NormalItem(item: Item) : UpdatableItem(item) {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: It would be even better and more structured to have separate classes for each case, that all inherit from UpdatableItem:

```
abstract class UpdatableItem(val item: Item) {
    abstract fun update()

    fun decreaseSellIn() {
        item.sellIn -= 1
    }

    fun increaseQuality(amount: Int = 1) {
        item.quality = minOf(50, item.quality + amount)
    }

    fun decreaseQuality(amount: Int = 1) {
        item.quality = maxOf(0, item.quality - amount)
    }
}

class NormalItem(item: Item) : UpdatableItem(item) {
    override fun update() {
        ///fill in function details here - you can access item properties using item.sellIn and item.quality
    }
}

class AgedBrie(item: Item) : UpdatableItem(item) {
    override fun update() {
        ///fill in function details here 
    }
}

class Sulfuras(item: Item) : UpdatableItem(item) {
    override fun update() {
        ///fill in function details here 
    }
}

class BackstagePass(item: Item) : UpdatableItem(item) {
    override fun update() {
        ///fill in function details here 
    }
}

class ConjuredItem(item: Item) : UpdatableItem(item) {
    override fun update() {
        ///fill in function details here 
    }
}
```

If you use this approach, in the GildedRose `updateQuality()` function (which I've asked you to put back in please) you'll then need to map each item to its class and then call its `update()` function. Hope that makes sense! :)

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt:1]: We need to keep the GildedRose class please:

```
class GildedRose(val items: List<Item>) {

    fun updateQuality() {
        //update all the items here
    }
}
```

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 19]
```diff
@@ -1,58 +1,49 @@
...
+    override fun update() {
+        val staticGoods = arrayOf("Sulfuras, Hand of Ragnaros")
+        if (staticGoods.contains(item.name)) maintainQuality()
+        else decreaseSellIn()
+        //    val appreciatingGoods = arrayOf("Aged Brie", "Backstage passes to a TAFKAL80ETC concert")
+        val depreciatingGoods = arrayOf("+5 Dexterity Vest", "Elixir of the Mongoose")
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: the intention is that a good with any name other than the special specified names should be treated as a regular good, not just these two :)

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 1 above)
Comment (AmeliaIsCoding) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: yes that's right, this should be the default type :)

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 20]
```diff
@@ -1,58 +1,49 @@
...
+        if (item.name == "Aged Brie") increaseQuality(1)
+        if (item.sellIn > 10 && item.name == "Backstage passes to a TAFKAL80ETC concert") increaseQuality(1)
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: this is a lot of if statements. I think this code will change if you follow my other suggestions, but for future reference, it would be preferable to use a when statement and then handle the logic for each individual case within its own case.

```
when {
    item.name == "Aged Brie" -> //handle aged brie
    item.name == "Sulfuras, Hand of Ragnaros" -> // handle sulfuras
    item.name.startsWith("Backstage passes") -> //handle backstage pass
    item.name.startsWith("Conjured") -> //handle conjured items
    else -> //handle normal item
}
```

hope this makes sense!

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt [hunk 21]
```diff
@@ -1,58 +1,49 @@
...
 }
 
+class NormalItem(item: Item) : UpdatableItem(item) {
+    override fun update() {
+        val staticGoods = arrayOf("Sulfuras, Hand of Ragnaros")
+        if (staticGoods.contains(item.name)) maintainQuality()
+        else decreaseSellIn()
+        //    val appreciatingGoods = arrayOf("Aged Brie", "Backstage passes to a TAFKAL80ETC concert")
+        val depreciatingGoods = arrayOf("+5 Dexterity Vest", "Elixir of the Mongoose")
+        if (item.name == "Aged Brie") increaseQuality(1)
```
Comment (AmeliaIsCoding) [inline Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: do you handle the expired aged brie case? I'm not sure I can see this

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 22]
```diff
@@ -6,14 +6,178 @@ import org.junit.jupiter.api.Test
...
+        val item = (Item("Aged Brie", 2, 0))
+        val app = NormalItem(item)
+        app.update()
+        assertEquals("Aged Brie", app.item.name)
+        assertEquals(1, app.item.sellIn)
+        assertEquals(1, app.item.quality)
+    }
+
+    @Test
+    fun `Aged brie quality increases by 1 after expiry`() {
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: aged brie should increase by 2 after expiry :)

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt [hunk 23]
```diff
@@ -6,14 +6,178 @@ import org.junit.jupiter.api.Test
...
+        val item = (Item("Aged Brie", 0, 0))
```
Comment (AmeliaIsCoding) [inline Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: do you need these parentheses outside `Item`? why not just `val item = Item("Aged Brie", 0, 0)` and similarly for other tests :)

**Review by AmeliaIsCoding — COMMENTED** [review]
good work! well done with using classes! also nice work on using git. I think there's still a lot to be done with the class structure, if you could read my comments and have a go at implementing the changes that would be great please :)

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 10 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: done

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 11 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: Very true, I've removed it. I kept it in before as I wasn't sure if despite it not doing anything it was good to have it there for completeness sake so that the handling of sulfuras is made explicit - I thought it would be preferable that way for readability.

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt (hunk 12 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt]: whoops! fixed now.

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt (hunk 13 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt:19]: done.

File: Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt (hunk 12 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/TexttestFixture.kt]: done

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 1 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: ok, that should now be the case : )

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 20 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: I think I've implemented this correctly now? Although cases like the backstage pass when im testing the sellIn dates (between 1 and 5 or 5 and 10 ... ) I'm still using if statements. I could change those to more whens but is it better to have nested whens than some more if statements? or have I confused the matter entirely and actually there should be a separate function per item type which each call on common functions like increaseQuality() ?

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt (hunk 21 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt]: added this in.

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt (hunk 22 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: OH sorry - I didn't realise, I've changed that now

File: Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt (hunk 23 above)
Comment (ClaraBollenGandolfo) [reply Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt]: removed those - not sure what they were a remnant of

File: Kotlin/local.properties
Comment (ClaraBollenGandolfo) [reply Kotlin/local.properties:1]: i'm unsure if I've done this correctly. I edited my .gitignore and pushed my updated code which meant those files from before weren't updated, but they're still in the PR because of the previous commits from when they weren't in .gitignore. Is this fine or have I done it wrong - should it have wiped all changes to the files even over previous commits?
